extensions/pyRevitCore.extension/pyRevit.tab/pyRevit.panel/Settings.smartbutton/script.py

```diff
@@ -389,42 +388,27 @@ def _setup_addinfiles(self):
             {str(x.Product.ProductYear):x
              for x in PyRevit.PyRevitAttachments.GetAttachments()}
 
-        for rvt_ver, checkbox in self._addinfiles_cboxes.items():
-            if rvt_ver in attachments:
-                if rvt_ver != HOST_APP.version:
-                    checkbox.Content = \
-                        self._make_product_name(
-                            attachments[rvt_ver].Product,
-                            ''
-                            )
+        for rvt_cb in self.supported_revit_versions_CB:
+            if rvt_cb.Version in attachments:
+                if rvt_cb.Version != HOST_APP.version:
+                    rvt_cb.Content = rvt_cb.Content + self._make_product_name(attachments[rvt_cb.Version].Product,'')
                 else:
-                    checkbox.Content = \
-                        self._make_product_name(
-                            attachments[rvt_ver].Product,
-                            self.get_locale_string("RevitAttachment.Current")
-                            )
-
-                checkbox.IsChecked = True
-                if attachments[rvt_ver].AttachmentType == \
-                        PyRevit.PyRevitAttachmentType.AllUsers:
-                    checkbox.IsEnabled = False
-                    checkbox.Content += self.get_locale_string("RevitAttachment.AllUsers")
+                    rvt_cb.Content = rvt_cb.Content + self._make_product_name(attachments[rvt_cb.Version].Product,self.get_locale_string("RevitAttachment.Current"))
```

Comment:
I know that this is not your code, but while you're at it... 😉 
the only thing that changes is the note, so we can introduce a variable for it and avoid the repeated code

```
note = "" if rvt_ver in attachments else self.get_locale_string("RevitAttachment.Current")
checkbox.Content += self._make_product_name(attachments[rvt_ver].Product, note)
```

we could even introduce a variable for `attachments[rvt_ver]` and use it also for the `AttachmentType` below